source/Playnite/GamesEditor.cs

```diff
@@ -467,6 +467,206 @@ public void OpenGameLocation(Game game)
             }
         }
 
+        public void CalculateGameSize(Game game, bool onlyIfDataMissing, bool updateGame = true, bool checkLastScanDate = false)
+        {
+            if (!game.IsInstalled)
+            {
+                return;
+            }
+
+            if (game.InstallSize != null && onlyIfDataMissing)
+            {
+                return;
+            }
+
+            try
+            {
+                long size = 0;
+                if (game.Roms.HasItems())
+                {
+                    var expandedGame = GetExpandedGameForRomsSizeScan(game);
+                    if (checkLastScanDate)
+                    {
+                        var addedAfterLastCheck = expandedGame.LastSizeScanDate == null || expandedGame.Added != null && expandedGame.Added > expandedGame.LastSizeScanDate;
+                        if (!addedAfterLastCheck && !expandedGame.Roms.Any(x => !x.Path.IsNullOrWhiteSpace()
+                            && FileSystem.FileExists(x.Path) && expandedGame.LastSizeScanDate != null
+                            && FileSystem.FileGetLastWriteTime(x.Path) > expandedGame.LastSizeScanDate))

```

Comment:
This will not work with playlist files correctly. You should imo get your list of ROM files to scan first (including playlist children) and then do size calculation and check if the calculation is needed or not.  See my other commend in this file.

Reply:
Done in https://github.com/JosefNemec/Playnite/pull/3017/commits/046a08dff6bffdf451441cf191364da80cbd526f